### PR TITLE
Clean Up Logging for Net Init

### PR DESCRIPTION
**Pull request by shelhamer**
- condense conditions by `LOG_IF`
- only log memory usage once after all tops are made

Comment to vote for the current memory usage logging if there's a use case out there.

### CODE REVIEW

File: src/caffe/net.cpp [hunk 1]
```diff
@@ -684,25 +660,22 @@ void Net<Dtype>::ForwardDebugInfo(const int layer_id) {
...
     const string& blob_name = blob_names_[top_id_vecs_[layer_id][top_id]];
     const Dtype data_abs_val_mean = blob.asum_data() / blob.count();
-    if (Caffe::root_solver()) {
-      LOG(INFO) << "    [Forward] "
-                << "Layer " << layer_names_[layer_id]
-                << ", top blob " << blob_name
-                << " data: " << data_abs_val_mean;
-    }
-  }
+    LOG_IF(INFO, Caffe::root_solver())
```
Comment (ronghanghu) [inline src/caffe/net.cpp]: You seem to have removed extra } here.

Comment (shelhamer) [inline src/caffe/net.cpp]: @ronghanghu fixed, thanks.

**Comment by seanbell** [comment]
I think it makes sense to not print memory usage, since it's not the total memory usage and thus leads to confusion such as #2387 and #3042.

**Comment by shelhamer** [comment]
Dropping the memory usage log line is fine by me. It does cause confusion
and I don't know that anyone is making use of it.

Evan Shelhamer

**Pull request merged**